SocketRocket/Internal/SRConstants.m

```diff
@@ -0,0 +1,19 @@
+//
+// Copyright (c) 2016-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree. An additional grant
+// of patent rights can be found in the PATENTS file in the same directory.
+//
+
+#import "SRConstants.h"
+
+size_t SRDefaultBufferSize(void) {
+    static size_t size;
+    static dispatch_once_t onceToken;
```

Comment:
Actually, in this case it's not an optimization.

On FreeBSD libc (or, apples fork of it: http://opensource.apple.com/source/Libc/Libc-1082.50.1/gen/FreeBSD/getpagesize.c), during the initial call to `getpagesize()`, they always make a syscall until they can cache the result. 

Now, AFIAK, `dispatch_once`, as it operates purely in userland would be faster than doing two syscalls required for `getpagesize()`.

(But note that libc does its own caching, which may just make it even better to have a direct call through to `getpagesize()` without a cache here).